### PR TITLE
Richard- Added Placeholders for Staff Pages & Implemented Staff CreatePage complete with tests and story

### LINKED ISSUE #16: Copy Placeholders for Staff pages; add to App.js/AppNavbar.js

### LINKED ISSUE #26: CreatePage for Staff plus tests and stories
Create a React component and route for the Create Page.
The user story this supports is:

As an instructor or admin
I can enter information about my course staff
So that I can delegate some functions to TAs/LAs

Acceptance Critieria:
- [ ] When you navigate to /courses/{id}/staff in the URL bar,
there is page that allows the user to create a new Staff record for the 
course in question.

- [ ] When you enter data into the form and click the submit button,
if the data is valid, the data is sent to the server,
a new Staff record is created for the course in question.

- [ ] When you enter data into the form and click the submit button,
if the data is not valid, appropriate error messages are displayed,
the data is NOT sent to the server, a new Staff record is NOT created.

- [ ] When you submit valid data to the page and click submit, a new valid database
entry is stored in the database.

### PR DESCRIPTION
**Pull request by richardfang888**
## Note

This is a rebased version of PR #27 which was asked to be done during section

---

In this PR, we add a React component for the creation of new course staff, complete with a story and test file. As a user, I can now go navigate to the route at /staff/create in the URL and enter data to create a new Staff record. Additionally, the staff index and staff edit pages have basic placeholder values which make them exist.

deployed at: https://organic-richardfang888.dokku-10.cs.ucsb.edu
Closes #26 
Close #16

### CODE REVIEW

**Review by SloggyYoggy — DISMISSED** [review]
LGTM! Good rebase

**Review by ukyl — DISMISSED** [review]
lgtm

image

**Review by abhishekambastha04 — DISMISSED** [review]
LGTM

File: frontend/src/App.js [hunk 1]
```diff
@@ -45,6 +49,14 @@ function App() {
     </>
   ) : null;
 
+  const staffRoutes = hasRole(currentUser, "ROLE_ADMIN") || hasRole(currentUser, "ROLE_INSTRUCTOR") ? (
+    <>
+      <Route path="/courses/:courseId/staff/create" element={<StaffCreatePage />} />
+      <Route path="/courses/:courseId/staff" element={<StaffIndexPage />} />
+      <Route path="/courses/:courseId/staff/edit/:id" element={<StaffEditPage />} />
```
Comment (pconrad) [inline frontend/src/App.js]:
```suggestion
```

There's no use case for a Staff Edit page.

File: frontend/src/main/pages/StaffEditPage.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+  // Stryker disable all : placeholder for future implementation
+  return (
+    <BasicLayout>
+      <div className="pt-2">
+        <h1>Edit page not yet implemented</h1>
+      </div>
+    </BasicLayout>
+  )
+}
```
Comment (pconrad) [inline frontend/src/main/pages/StaffEditPage.js]: Remove file... there's no use case.

File: frontend/src/App.js (hunk 1 above)
Comment (pconrad) [reply frontend/src/App.js]: You are either a staff member for a certain course, or you aren't.   So we only have add and delete.

**Review by pconrad — CHANGES_REQUESTED** [review]
Looks good, but we need to remove the edit function.

File: frontend/src/main/pages/StaffEditPage.js (hunk 2 above)
Comment (richardfang888) [reply frontend/src/main/pages/StaffEditPage.js]: Removed

File: frontend/src/App.js (hunk 1 above)
Comment (richardfang888) [reply frontend/src/App.js]: Removed route

**Pull request closed without merging**